arm/subscriptions/client.go

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator 0.11.0.0
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package subscriptions
+
+import (
+	"fmt"
+	"github.com/azure/go-autorest/autorest"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	ApiVersion             = "2014-04-01-preview"
+	DefaultBaseUri         = "https://management.azure.com"
+	DefaultPollingDuration = 10 * time.Minute
+)
+
+type SubscriptionClient struct {
+	autorest.Client
+	BaseUri        string
+	SubscriptionId string
+}
+
+func New(subscriptionId string) *SubscriptionClient {
+	return NewWithBaseUri(DefaultBaseUri, subscriptionId)
+}
+
+func NewWithBaseUri(baseUri string, subscriptionId string) *SubscriptionClient {
+	client := &SubscriptionClient{BaseUri: baseUri, SubscriptionId: subscriptionId}
+	client.PollingMode = autorest.PollUntilDuration
+	client.PollingDuration = DefaultPollingDuration
+	return client
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// Subscriptions Client
+//
+////////////////////////////////////////////////////////////////////////////////
+type SubscriptionsClient struct {
```

Comment:
@ahmetalpbalkan It is an artifact of generation. The generator created both from the names provided by the Swagger. The outer name is derived from the service (in this, case Subscriptions) and the inner name comes from the Swagger (see https://github.com/Azure/azure-rest-api-specs/blob/master/arm-subscriptions/2014-04-01-preview/swagger/subscriptions.json).